src/services/reportCache.js

```diff
@@ -164,6 +166,61 @@ const cacheResources = async (objectiveId, activityReportObjectiveId, resources
   ]);
 };
 
+const cacheCourses = async (objectiveId, activityReportObjectiveId, courses = []) => {
+  const courseIds = courses.map((course) => course.courseId);
+  const courseSet = new Set(courseIds);
+  const originalAroCourses = await ActivityReportObjectiveCourse.findAll({
+    where: { activityReportObjectiveId },
+    raw: true,
+  });
+  const originalCourseIds = originalAroCourses.map((course) => course.courseId)
+    || [];
+  const removedCourseIds = originalCourseIds.filter((id) => !courseSet.has(id));
+  const currentCourseIds = new Set(originalCourseIds.filter((id) => courseSet.has(id)));
+  const newCourseIds = courseIds.filter((id) => !currentCourseIds.has(id));
+
+  return Promise.all([
+    ...newCourseIds.map(async (courseId) => ActivityReportObjectiveCourse.create({
+      activityReportObjectiveId,
+      courseId,
+    })),
+    removedCourseIds.length > 0
+      ? ActivityReportObjectiveCourse.destroy({
+        where: {
+          activityReportObjectiveId,
+          courseId: { [Op.in]: removedCourseIds },
+        },
+        individualHooks: true,
+        hookMetadata: { objectiveId },
+      })
+      : Promise.resolve(),
+    newCourseIds.length > 0
+      ? ObjectiveCourse.update(
+        { onAR: true },
+        {
+          where: { courseId: { [Op.in]: newCourseIds } },
+          include: [
+            {
+              model: Objective,
+              as: 'objective',
+              required: true,
+              where: { id: objectiveId },
+              include: [
+                {
+                  model: ActivityReportObjective,
+                  as: 'activityReportObjectives',
+                  required: true,
+                  where: { id: activityReportObjectiveId },
```

Comment:
Does this work join on update. I know we have had issues in the past.

Reply:
don't need this code, removed